static/js/containers/CollectionListPage.js

```diff
@@ -65,13 +54,20 @@ class CollectionListPage extends React.Component {
     </ul>;
   }
 
+  openNewCollectionDialog = (e: MouseEvent) => {
+    const { dispatch } = this.props;
+
+    e.preventDefault();
+    dispatch(collectionUiActions.startNewCollectionDialog());
+  };
+
   render() {
     const { commonUi } = this.props;
     const formLink = SETTINGS.editable
       ? (
-        <a className="button-link create-collection-button" href="/collection_form">
-          <i className="material-icons">add</i>
-          Create a Collection
+        <a href="#" className="button-link create-collection-button" onClick={this.openNewCollectionDialog}>
```

Comment:
Just curious: what's the reasoning behind the `href` addition? Seems like we need `preventDefault` only because of that.